### PR TITLE
Allow multiple windows

### PR DESCRIPTION
**Pull request by miherlosev**
Changes:
* pass flag `allowMultipleWindows` to the `domProcessor` and `Session`
* allow `FormData` as body of the service messages
* opening a link in a new window
* special mechanism for handling form submitting (prevent form submitting on client, send form content to the server, open form action url with special id allowed to show result of form submitting on the opened page) - partial support only for form.enctype= `application/x-www-form-urlencoded`.
* add `multi-window-http-playground` gulp task.

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/client/sandbox/native-methods.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ const NATIVE_CODE_RE = /\[native code]/;
 
 class NativeMethods {
     isStoragePropsLocatedInProto: boolean;
-    createDocumentFragment: any;
-    createElement: any;
-    createElementNS: any;
+    createDocumentFragment: () => DocumentFragment;
+    createElement: (tagName: string) => HTMLElement;
+    createElementNS: (namespaceURI: string, qualifiedName: string) => HTMLElement;
```
Comment (LavrovArtem) [inline src/client/sandbox/native-methods.ts]:
```ts
    createDocumentFragment: Document['createDocumentFragment'];
    createElement: Document['createElement'];
    createElementNS: Document['createElementNS'];
```

File: src/client/sandbox/node/element.ts [hunk 2]
```diff
@@ -913,7 +956,28 @@ export default class ElementSandbox extends SandboxBase {
...
+
+            const form = e.target;
+
+            if (!this._shouldOpenInNewWindow(form.target))
+                return;
+
+            const aboutBlankUrl = urlUtils.getProxyUrl(SPECIAL_BLANK_PAGE);
+            const windowName    = this._openUrlInNewWindow(aboutBlankUrl);
+
+            form.target = windowName;
```
Comment (LavrovArtem) [inline src/client/sandbox/node/element.ts]: May be something like that:
```ts
const storedTarget = form.target;

form.target = windowName;

form.submit();

form.target = storedTarget;

preventEvent();
stopEventPropagation();
```

File: src/client/sandbox/native-methods.ts [hunk 3]
```diff
@@ -127,7 +127,7 @@ class NativeMethods {
     WindowTextEvent: any;
     WindowMouseEvent: any;
     canvasContextDrawImage: any;
-    formDataAppend: any;
+    formDataAppend: (name: string, value: string | Blob, fileName?: string) => void;
```
Comment (LavrovArtem) [inline src/client/sandbox/native-methods.ts]:
```ts
    formDataAppend: FormData['append'];
```

File: src/client/sandbox/native-methods.ts [hunk 4]
```diff
@@ -313,6 +313,7 @@ class NativeMethods {
     Worker: any;
     ArrayBuffer: any;
     Uint8Array: any;
+    Uint32Array: Function;
```
Comment (LavrovArtem) [inline src/client/sandbox/native-methods.ts]:
```ts
    Uint32Array: Uint32Array['constructor']
```

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**